cmd/root/bundle.go

```diff
@@ -81,6 +81,19 @@ func configureBundle(cmd *cobra.Command, b *bundle.Bundle) (*bundle.Bundle, diag
 
 	// Configure the workspace profile if the flag has been set.
 	diags = diags.Extend(configureProfile(cmd, b))
+	if diags.HasError() {
+		return b, diags
+	}
+
+	// Set the auth configuration in the command context. This can be used
+	// downstream to initialize a API client.
+	client, err := b.InitializeWorkspaceClient()
```

Comment:
There is also WorkspaceClient() which creates and caches the client on bundle. 

If that's not necessary, should we clean that up?

Related: Bundle.client, Bundke.clientOnce, can we get rid of those?

Reply:
> If that's not necessary, should we clean that up?

Eyeballing it we could clean that up a bit without issues. It's out of the scope of what this PR is trying to achieve, though, IMO.

Reply:
I think it's in scope, we should not have multiple ways to set up client.

Also, please take a look "variables v2" doc. How would this work if we want to interpolate variables first?

Reply:
> we should not have multiple ways to set up client.

Fair enough. Modified the change to set the client as well. I was a bit hesitant to do this because it's not ideal how decoupled the setting of the client is from the usage of it but this change should be safe enough (and covered by tests).

> How would this work if we want to interpolate variables first?

We'd need to resolve the variables before we resolve the client.

Reply:
Changed the approach to still ensure that the client is only initialized / set once. Did so by renaming `InitializeWorkspaceClient` to `WorkspaceClientE`